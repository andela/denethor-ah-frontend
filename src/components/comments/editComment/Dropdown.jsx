import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class Dropdown extends Component {
  componentDidMount () {
    document.addEventListener('click', this.handleToggle);
  }

  componentWillUnmount () {
    document.removeEventListener('click', this.handleToggle);
  }

  handleToggle = () => {
    this.props.toggleModal();
  };

  render() {
    const { isAuthor, toggleMode, showHistory, hasHistory } = this.props;

    return (
      <div className='comment__edit-comment__drop-down' ref={node => this.modal = node}>
        <ul>
          {isAuthor && <li><button onClick={() => { toggleMode('edit') }}>Edit</button></li>}
          {
            hasHistory
              && (
                <li>
                  <button onClick={() => showHistory(true)}>
                    View History
                  </button>
                </li>
              )
          }
        </ul>
      </div>
    );
  }
}

Dropdown.propTypes = {
  isAuthor: PropTypes.bool,
  toggleModal: PropTypes.func,
  toggleMode: PropTypes.func,
  showHistory: PropTypes.func,
  hasHistory: PropTypes.object
}

export default Dropdown;
