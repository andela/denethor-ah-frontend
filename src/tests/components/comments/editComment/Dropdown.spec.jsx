import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from '../../../../components/comments/editComment/Dropdown';

it('Should render static component', () => {
  const wrapper = shallow(<Dropdown isAuthor={true} hasHistory={true} />);
  expect(wrapper).toMatchSnapshot();
});

it('Should call handle functions onclick', () => {
  const toggleMode = jest.fn();
  const showHistory = jest.fn();
  const props = {
    toggleMode, showHistory, isAuthor: true, hasHistory: true
  }
  const wrapper = shallow(<Dropdown {...props} />);
  wrapper.find('button').at(0).simulate('click');
  expect(toggleMode).toHaveBeenLastCalledWith('edit');
  wrapper.find('button').at(1).simulate('click');
  wrapper.unmount();
  expect(showHistory).toHaveBeenLastCalledWith(true);
});
