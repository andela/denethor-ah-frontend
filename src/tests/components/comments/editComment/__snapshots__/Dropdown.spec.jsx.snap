// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render static component 1`] = `
<div
  className="comment__edit-comment__drop-down"
>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        Edit
      </button>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        View History
      </button>
    </li>
  </ul>
</div>
`;
